src/Microsoft.AspNet.Mvc.TagHelpers/ImageTagHelper.cs

```diff
@@ -61,18 +63,23 @@ public ImageTagHelper(IHostingEnvironment hostingEnvironment, IMemoryCache cache
 
         protected IMemoryCache Cache { get; }
 
+        protected IUrlHelper UrlHelper { get; }
+
         /// <inheritdoc />
         public override void Process(TagHelperContext context, TagHelperOutput output)
         {
             if (AppendVersion)
             {
                 EnsureFileVersionProvider();
+                Src = UrlHelper.Content(Src.Trim());
```

Comment:
Why restore surrounding whitespace in the `UrlResolutionTagHelper` tag helper but drop it in the others?

Also `Trim()` has the same Unicode vs. HTML issue as `IsWhiteSpace()`. Should call it as `Src.Trim(ValidAttributeWhitespaceChars)` after moving `ValidAttributeWhitespaceChars` some place central.